src/compiler/checker.ts

```diff
@@ -1285,13 +1291,29 @@ namespace ts {
                 }
                 return undefined;
             }
+
+            const patternModuleSymbol = getPatternAmbientModule(moduleName);
+            if (patternModuleSymbol) {
+                return getMergedSymbol(patternModuleSymbol);
+            }
+
             if (moduleNotFoundError) {
                 // report errors only if it was requested
                 error(moduleReferenceLiteral, moduleNotFoundError, moduleName);
             }
             return undefined;
         }
 
+        /** Get an ambient module with a wildcard ("*") in it. */
+        function getPatternAmbientModule(name: string): Symbol | undefined {

```

Comment:
nit. i would inline this function. do not see much cognitive savings from pulling it out.